src/com/amazon/corretto/crypto/provider/EvpEcPrivateKey.java

```diff
@@ -23,6 +24,17 @@ class EvpEcPrivateKey extends EvpEcKey implements ECPrivateKey, CanDerivePublicK
         super(key, false);
     }
 
+    @Override
+    public boolean equals(final Object obj) {
+        final boolean parentEquals = super.equals(obj);
+        if (parentEquals) {
+            return parentEquals;
+        } else if (Utils.getJavaVersion() == 10 && (obj instanceof ECPrivateKey)) {
+            return this.getS().equals(((ECPrivateKey) obj).getS());
+        }
+        return parentEquals;
+    }
```

Comment:
i'm not thrilled about this override.

on one hand, it might be useful that we have a shared notion of equality with JDK10 here for _some_ users (although JDK10 is long-since EOL'd and shouldn't be used).

on the other, it feels like we're taking a half-measure towards compatibility. we're gaining compatibility with JDK10 on object equality, but there's nothing we can do about compatibility on `hashCode` without DER-decoding the bytes, and re-encoding them having stripped the leading `0`-valued byte, which feels _very much_ like the wrong thing to do -- both for performance and because it accedes to a model of "bug-for-bug" compatibility that gets ugly quickly.

Reply:
after digging into JDK history a bit more, i backed this override out. we need to prioritize java as it is, not java as it was.